safehttp/config.go

```diff
@@ -16,9 +16,9 @@ package safehttp
 
 // Config provide additional configurations to Interceptors when needed.
 type Config interface {
-	// Apply applies the configurations to an Interceptor if it's of the
+	// Apply applies the configurations to an Interceptor, if it's of the
 	// appropriate type. The bool indicates whether the configuration took
-	// effect. If that's the case, the modified interceptor should be returned
-	// and otherwise Apply should return a nil Interceptor.
+	// effect, in which case, the modified interceptor will be returned.
+	// Otherwise, Apply returns the unmodified Interceptor.
```

Comment:
Maybe just do not specifying what Interceptor is when ok==false. Another question that arises is why we need to have OK at all. Why isn't `nil` a fine contract to say "I couldn't do anything with this type". What's the benefit of the bool? The implementation ignores the returned Interceptor anyway, if ok==false.

Also is it possible that Apply returns an error? What would an implementation do if Config results in an invalid Interceptor ... or should this be handled when creating the Config, so it shouldn't be Handler registrations business?

Reply:
We have now removed `Apply`.